### ISSUE TITLE
CustomImportOrder check: SAME_PACKAGE(n) works incorrectly

### ISSUE THREAD

**Issue by ivanov-alex** [checkstyle/checkstyle#1262]
There is a problem with SAME_PACKAGE(n) group. In example below you can see that it tends to include imports from packages that are higher in hierarchy into the SAME_PACKAGE group.
```java
package java.util.concurrent.locks;
///// All the imports #1-8 are CORRECTLY treated as a one group in case of SAME_PACKAGE(2)
// Imports #1-6 are INCORRECTLY treated as "same package group" in case of SAME_PACKAGE(3) or SAME_PACKAGE(4)
// In fact only #4-6 should be on SAME_PACKAGE(3), only #6 on SAME_PACKAGE(4)
import java.util.*; //#1
import java.util.List; //#2
import java.util.StringTokenizer; //#3
import java.util.concurrent.*; //#4
import java.util.concurrent.AbstractExecutorService; //#5
import java.util.concurrent.locks.LockSupport; //#6
//These 2 imports are CORRECTLY treated as "no group" in case of SAME_PACKAGE(3)
import java.util.regex.Pattern; //#7
import java.util.regex.Matcher; //#8

public class InputCustomImportOrderSamePackage2 {
}
```

Configuration file:
```xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC "-//Puppy Crawl//DTD Check Configuration 1.3//EN" "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
<module name = "Checker">
    <property name="charset" value="UTF-8"/>
    <property name="severity" value="warning"/>
    <module name="TreeWalker">
	    <module name="CustomImportOrder">
	        <property name="customImportOrderRules" value="SAME_PACKAGE(4)"/>
	        <property name="sortImportsInGroupAlphabetically" value="true"/>
	        <property name="separateLineBetweenGroups" value="true"/>
	    </module>
    </module>
</module>
```
